cql3/restrictions/multi_column_restriction.hh

```diff
@@ -376,17 +376,24 @@ protected:
 };
 
 class multi_column_restriction::slice final : public multi_column_restriction {
-    using restriction_shared_ptr = ::shared_ptr<clustering_key_restrictions>;
+public:
+    enum class mode : char {
+        normal,
+        scylla_clustering_bound,
+    };
```

Comment:
@dekimir doesn't this need to go via expressions?

Reply:
Not yet.  Expressions are used only in filtering ATM, not in creating the clustering slice.

(If curious, see [this branch](https://github.com/dekimir/scylla/commits/drop-restrictions-class.v0) for WIP on how expressions will handle clustering slice.)

Reply:
To be more precise: expressions are used everywhere the old `restriction` class was used, including some aspects of creating the partition and clustering ranges.  But the `restrictions` (plural) class remains, and this code is a part of that hierarchy: `multi_column_restriction::slice` : `multi_column_restriction` : `primary_key_restrictions` : `restrictions`.

As indicated above, I'm working on replacing that hierarchy with simple expression conjunctions.